src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/LocalAppContextSwitches.cs

```diff
@@ -78,13 +78,13 @@ public static bool UseCompatibilityProcessSni
             {
                 if (s_useCompatibilityProcessSni == Tristate.NotInitialized)
                 {
-                    if (AppContext.TryGetSwitch(UseCompatibilityProcessSniString, out bool returnedValue) && returnedValue)
+                    if (AppContext.TryGetSwitch(UseCompatibilityProcessSniString, out bool returnedValue) && !returnedValue)
                     {
-                        s_useCompatibilityProcessSni = Tristate.True;
+                        s_useCompatibilityProcessSni = Tristate.False;
                     }
                     else
                     {
-                        s_useCompatibilityProcessSni = Tristate.False;
+                        s_useCompatibilityProcessSni = Tristate.True;
```

Comment:
@ErikEJ in this PR the default is changed to that UseCompatibilityProcessSni is true by default. That means that new async behaviour is false by default.

Reply:
Thanks, so in a repro context, we should use:

```c#
AppContext.SetSwitch("Switch.Microsoft.Data.SqlClient.UseCompatibilityAsyncBehaviour", false);
AppContext.SetSwitch("Switch.Microsoft.Data.SqlClient.UseCompatibilityProcessSni", false);

```

Reply:
For public consumption the settings you give are the defaults. 

If you can find an issue I want to know about it so I can fix it, regardless of the settings but you will need to tell me what the settings are so I can try to repro.